### PR TITLE
Format

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: sign.js [hunk 1]
```diff
@@ -38,57 +34,59 @@ function sign (hash, key, hashType, signType) {
...
 
+  var curve = new EC(curveId)
   var key = curve.genKeyPair()
+
   key._importPrivate(priv.privateKey)
   var out = key.sign(hash)
+
   return new Buffer(out.toDER())
 }
-function dsaSign (hash, priv, algo) {
```
Comment (dcousens) [inline sign.js]: @calvinmetcalf are these meant to be here?  They are completely unused.

File: sign.js [hunk 2]
```diff
@@ -144,31 +144,37 @@ function bits2octets (bits, q) {
...
-function makeR (g, k, p, q) {
-  return g.toRed(BN.mont(p)).redPow(k).fromRed().mod(q)
+      .update(kv.v)
+      .digest()
+  } while (k.cmp(q) !== -1)
+
+  return k
 }
+
+// function makeR (g, k, p, q) {
```
Comment (dcousens) [inline sign.js]: As is this, unused.

File: verify.js [hunk 3]
```diff
@@ -77,19 +78,19 @@ function dsaVerify (sig, hash, pub) {
   var r = unpacked.r
   checkValue(s, q)
   checkValue(r, q)
-  var montq = BN.mont(q)
```
Comment (dcousens) [inline verify.js]: Unused

**Comment by dcousens** [comment]
Failing test, seems to be something wrong with `parse-asn1`?

```
/home/travis/build/crypto-browserify/browserify-sign/sign.js:23
  var len = priv.modulus.byteLength()
                         ^
TypeError: Cannot call method 'byteLength' of undefined
```

File: sign.js [hunk 4]
```diff
@@ -1,29 +1,25 @@
...
   var priv = parseKeys(key)
   if (priv.curve) {
-    if (signType !== 'ecdsa') {
-      throw new Error('wrong private key type')
-    }
+    if (signType !== 'ecdsa') throw new Error('wrong private key type')
+
     return ecSign(hash, priv)
   } else if (priv.type === 'dsa') {
-    return dsaSign(hash, priv, hashType)
```
Comment (calvinmetcalf) [inline sign.js]: this is where dsaSign was used from

**Comment by calvinmetcalf** [comment]
here we go with a version that passes standard and has DSA in it

**Comment by calvinmetcalf** [comment]
everything passes, so merging

File: sign.js (hunk 4 above)
Comment (dcousens) [inline sign.js]: Cheers for re-adding it.  Must have been lost when I re-based.

**Comment by dcousens** [comment]
Thanks @calvinmetcalf,  not sure how that `dsaSign` got lost.

**Comment by calvinmetcalf** [comment]
merge I'd guess, but that's why we use standard know

**Comment by calvinmetcalf** [comment]
and published

**Pull request merged**